### ISSUE TITLE
query with like and percent sign

### ISSUE THREAD

**Issue by DJafari** [vitusortner/floor#154]
how can run this query with floor ?

```sql
SELECT * FROM cities WHERE name LIKE '%U%'
```

 when i using this :

```sql
SELECT * FROM cities WHERE name LIKE %:search%
```

after rebuild, in DataBase.g.part :

```dart
return _queryAdapter.queryList('SELECT * FROM cities WHERE name LIKE %?',
        arguments: <dynamic>[search], mapper: _buyerMapper);
```

**Comment by vitusortner** [comment]
It's possible to enclose the `%` characters in the input query but not in the query itself. An example can be seen in the following.

```dart
// dao
@Query('SELECT * FROM cities WHERE name LIKE :search')
Future<List<City>> findCitiesWithNameLike(String search);

// usage
final search = '%U%';
await dao.findCitiesWithNameLike(search);
```

**Comment by tulioccalazans** [comment]
Hi @vitusortner ,
I found a bug involving LIKE operation.

This does not work:
```
@Query('SELECT * FROM Contact WHERE IsAssistant = 1 AND Code != :notUser AND Name LIKE :text ORDER BY `Order` ASC')
  Future<List<Contact>> searchAssistantsByText(String text, String notUser);
```
But it work:
```
@Query('SELECT * FROM Contact WHERE IsAssistant = 1 AND Name LIKE :text AND Code != :notUser ORDER BY `Order` ASC')
  Future<List<Contact>> searchAssistantsByText(String text, String notUser);
```
Note that the unique difference is **AND Code != :notUser** order
Maybe it is related with the operator **!=**

**Comment by vitusortner** [comment]
Floor is currently mapping parameters into the query in the same order as they are passed into the function. This issue will be fixed with the 1.0 release. More information and related issues: #142 #224

@tulioccalazans
